app/models/work.rb

```diff
@@ -0,0 +1,41 @@
+class Work < ApplicationRecord
+  # QUESTION: how to get this to work? CATEGORIES = ["album", "book", "movie"]
+  has_many :votes, dependent: :delete_all
+
+  validates :title, presence: true, uniqueness: {scope: :category, message: "This category already knows about that title." }
+  # TODO: validates :publication_year --> how to get year only??
+
+  # TODO: necessary???? validates_inclusion_of :category, :in => CATEGORIES
+
+  def self.work_categories
+    @categories = ["album", "book", "movie"]
+  end
+
+  def self.categorize_works(work_category)
+    return self.where(category: work_category)
+  end
+
+  def self.sort_works(work_category)
+    cat_works_array = categorize_works(work_category)
+    works_by_vote = cat_works_array.sort_by{|work| work.count_votes}.reverse
+    return works_by_vote
```

Comment:
I love how the methods in this file build upon one another, each adding a small amount of functionality. This is an excellent example of functional decomposition.